apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java

```diff
@@ -188,6 +189,84 @@ public void mapEobCommonGroupInpOutHHAHospiceSNFWhenNoFiNumberExpectNoFiNumExten
     assertNull(fiNumExtension);
   }
 
+  /**
+   * Verifies that {@link
+   * gov.cms.bfd.server.war.r4.providers.TransformerUtilsV2#careTeamHasMatchingExtension(
+   * (org.hl7.fhir.r4.model.ExplanationOfBenefit.CareTeamComponent, String, String)} verifies if an
+   * extension is found
+   */
+  @Test
+  public void careTeamHasMatchingExtensionReturnsTrueWhenFound() {
+    String referenceUrl = "http://test.url";
+    String codeValue = "code";
+    Coding coding = new Coding();
+    coding.setCode(codeValue);
+    Extension extension = new Extension(referenceUrl);
+    extension.setValue(coding);
+    CareTeamComponent careTeamComponent = new CareTeamComponent();
+    careTeamComponent.addExtension(extension);
+
+    boolean returnResult =
+        TransformerUtilsV2.careTeamHasMatchingExtension(careTeamComponent, referenceUrl, codeValue);
+
+    assertTrue(returnResult);
+  }
+
+  /**
+   * Verifies that {@link
+   * gov.cms.bfd.server.war.r4.providers.TransformerUtilsV2#careTeamHasMatchingExtension(
+   * (org.hl7.fhir.r4.model.ExplanationOfBenefit.CareTeamComponent, String, String)} verifies it
+   * returns false when a reference url is null or empty.
+   */
+  @Test
+  public void careTeamHasMatchingExtensionReturnsFalseWithNullOrEmptyReferenceUrl() {
+    String referenceUrl = "";
+    String codeValue = "code";
+    Coding coding = new Coding();
+    coding.setCode(codeValue);
+    Extension extension = new Extension(referenceUrl);
+    extension.setValue(coding);
+    CareTeamComponent careTeamComponent = new CareTeamComponent();
+    careTeamComponent.addExtension(extension);
+
+    boolean returnResult =
+        TransformerUtilsV2.careTeamHasMatchingExtension(careTeamComponent, referenceUrl, codeValue);
+
+    assertFalse(returnResult);
+
+    TransformerUtilsV2.careTeamHasMatchingExtension(careTeamComponent, null, codeValue);
+
+    assertFalse(returnResult);
+  }
+
+  /**
+   * Verifies that {@link
+   * gov.cms.bfd.server.war.r4.providers.TransformerUtilsV2#careTeamHasMatchingExtension(
+   * (org.hl7.fhir.r4.model.ExplanationOfBenefit.CareTeamComponent, String, String)} verifies it
+   * returns false when a code valueis null or empty.
+   */
+  @Test
+  public void careTeamHasMatchingExtensionReturnsFalseWithNullOrEmptyCodeValue() {
```

Comment:
Shouldn't we have separate tests for null vs empty string; a null could trigger an NPE depending on underlying code.

Reply:
done